assets/js/arrayProcessingTool.js

```diff
@@ -0,0 +1,94 @@
+const inputArrayHTMLElement=document.getElementById('array');
+const resultHTMLElement=document.getElementById('result');
+const searchButtonHTMLElement=document.getElementById('search-button');
+const sequenceButtonHTMLElement=document.getElementById('sequence-button');
+const maximalSumButtonHTMLElement=document.getElementById('maximal-sum-button');
+
+inputArrayHTMLElement.onchange=()=>{
+  object.arr=inputArrayHTMLElement.value.split(' ').map(el=>Number(el));
+  console.log(object.arr);
+  if(object.arr.includes(NaN)){
+      resultHTMLElement.style.backgroundColor='#f85454'
+      resultHTMLElement.textContent='some element(s) in array is not correct. Result could fail';
+  }
+};
+
+const object={
+    arr:[],
+    subSum: function (){
+        let sum = 0;
+        let currentSum=0;
+        for(let i=0;i<this.arr.length;i++) {
+            currentSum = 0;
+            for (let j = 0; j < this.arr.length && i + j < this.arr.length; j++) {
+                console.log(this.arr[i], this.arr[i + j], currentSum)
+                if (this.arr[i] < 0) {
+                    break;
+                }
+                if ((this.arr[i + j] < 0) && (Math.abs(this.arr[i + j]) > currentSum)) {
+                    i += j;
+                    break;
+                } else {
+                    currentSum += this.arr[i + j];
+                }
+            }
+            if (currentSum > sum) {
+                sum = currentSum;
+            }
+            return sum;
+        }
+    },
+    search:function (){
```

Comment:
The program returns an invalid median for the next input data:

- 2 3 1

Please, look at the definition of the median ([link](https://ru.wikipedia.org/wiki/%D0%9C%D0%B5%D0%B4%D0%B8%D0%B0%D0%BD%D0%B0_(%D1%81%D1%82%D0%B0%D1%82%D0%B8%D1%81%D1%82%D0%B8%D0%BA%D0%B0)) )

Reply:
fixed